### PR TITLE
added a troubleshooting section for macOS mojave

### PR DESCRIPTION
**Pull request by dustinlahr**
I ran into this issue recently trying out crystal and hoped to contribute!

[crystal-7209](https://github.com/crystal-lang/crystal/issues/7209)

### CODE REVIEW

**Comment by straight-shoota** [comment]
Other relevant issue: crystal-lang/crystal#6875

**Comment by RX14** [comment]
Can't someone who uses homebrew just work out a solution to make homebrew package crystal properly? If it doesn't work out the box, it's a packaging problem, not crystal's problem.

**Comment by proyb6** [comment]
@codingdracula I checked yesterday pkg-config wasn’t installed by default

**Comment by codingdracula** [comment]
@proyb6 apologies, everyone should ignore the last part of my post. I didn't realize it was still open

**Comment by straight-shoota** [comment]
@codingdracula Thank you! Your contribution is very welcome. I think we should definitely merge this as a service to Crystal users in MacOS. Unless there is a way to avoid this problem altogether as @RX14 asked. That would be even better, if no troubleshooting was required in the first place.

**Comment by RX14** [comment]
In the short term we should merge this PR, but only *after* we've bought up this issue with homebrew, or the issue will end up being neglected.

File: installation/on_mac_osx_using_homebrew.md [hunk 1]
```diff
@@ -21,3 +21,21 @@ you need to reinstall the command line tools and then select the default active
 $ xcode-select --install
 $ xcode-select --switch /Library/Developer/CommandLineTools
 ```
+
+## Troubleshooting on macOS MacOS 10.14.2 (Mojave)
```
Comment (bcardiff) [inline installation/on_mac_osx_using_homebrew.md]:
```suggestion
## Troubleshooting on macOS 10.14.2 (Mojave)
```

File: installation/on_mac_osx_using_homebrew.md [hunk 2]
```diff
@@ -21,3 +21,21 @@ you need to reinstall the command line tools and then select the default active
...
+
+if you get an error like:
```
Comment (bcardiff) [inline installation/on_mac_osx_using_homebrew.md]:
```suggestion
If you get an error like:
```

File: installation/on_mac_osx_using_homebrew.md [hunk 3]
```diff
@@ -21,3 +21,21 @@ you need to reinstall the command line tools and then select the default active
...
+
+you may need to install OpenSSL and link pkg-config to OpenSSL
+
+```
+brew install openssl
+```
+
+```
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/opt/openssl/lib/pkgconfig
+```
```
Comment (bcardiff) [inline installation/on_mac_osx_using_homebrew.md]: I would append something like:


> As with other keg-only formulas there are some caveats shown in `brew info <formula>` that advertise how to link pkg-config with this library.
> 
> Crystal compiler by default will use pkg-config to find the location of libraries to link.

**Comment by gregors** [comment]
@dustinlahr thanks for helping me solve my problem

**Comment by straight-shoota** [comment]
@dustinlahr Could you apply the suggestions from @bcardiff ?

File: installation/on_mac_osx_using_homebrew.md [hunk 4]
```diff
@@ -21,3 +21,27 @@ you need to reinstall the command line tools and then select the default active
...
+
+## Troubleshooting on macOS MacOS 10.14.2 (Mojave)
+
+If you get an error like:
+
+```
+ld: library not found for -lssl (this usually means you need to install the development package for libssl)
+```
+
+you may need to install OpenSSL and link pkg-config to OpenSSL
```
Comment (r00ster91) [inline installation/on_mac_osx_using_homebrew.md]:
```suggestion
you may need to install OpenSSL and link pkg-config to OpenSSL:
```

File: installation/on_mac_osx_using_homebrew.md [hunk 5]
```diff
@@ -21,3 +21,27 @@ you need to reinstall the command line tools and then select the default active
...
+```
+
+```
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/opt/openssl/lib/pkgconfig
+```
+
+```
+As with other keg-only formulas there are some caveats shown in brew info <formula> that advertise how to link pkg-config with this library.
+
+Crystal compiler by default will use pkg-config to find the location of libraries to link.
```
Comment (r00ster91) [inline installation/on_mac_osx_using_homebrew.md]:
```suggestion
The Crystal compiler will by default use pkg-config to find the locations of libraries to link with.
```

File: installation/on_mac_osx_using_homebrew.md [hunk 6]
```diff
@@ -21,3 +21,27 @@ you need to reinstall the command line tools and then select the default active
...
+
+```
+brew install openssl
+```
+
+```
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/opt/openssl/lib/pkgconfig
+```
+
+```
```
Comment (r00ster91) [inline installation/on_mac_osx_using_homebrew.md]: This is not code.
```suggestion
```

File: installation/on_mac_osx_using_homebrew.md [hunk 7]
```diff
@@ -21,3 +21,27 @@ you need to reinstall the command line tools and then select the default active
...
+As with other keg-only formulas there are some caveats shown in brew info <formula> that advertise how to link pkg-config with this library.
+
+Crystal compiler by default will use pkg-config to find the location of libraries to link.
+```
```
Comment (r00ster91) [inline installation/on_mac_osx_using_homebrew.md]:
```suggestion
```

File: installation/on_mac_osx_using_homebrew.md [hunk 8]
```diff
@@ -21,3 +21,27 @@ you need to reinstall the command line tools and then select the default active
...
+brew install openssl
+```
+
+```
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/opt/openssl/lib/pkgconfig
+```
+
+```
+As with other keg-only formulas there are some caveats shown in brew info <formula> that advertise how to link pkg-config with this library.
```
Comment (r00ster91) [inline installation/on_mac_osx_using_homebrew.md]:
```suggestion
As with other keg-only formulas there are some caveats shown in `brew info <formula>` that shows how to link pkg-config with this library.
```

Comment (r00ster91) [reply installation/on_mac_osx_using_homebrew.md]: @dustinlahr was the commit batch feature not available? It allows you to squash all suggestions into one commit.

Comment (dustinlahr) [reply installation/on_mac_osx_using_homebrew.md]: I did not see it, so I just did it individually.

Comment (r00ster91) [reply installation/on_mac_osx_using_homebrew.md]: No worries!

**Comment by dkniffin** [comment]
Not sure if it belongs as part of this PR or another one, but this page on the website needs to be updated as well: https://luckyframework.org/guides/installing/

Right now it says:
> If you are on macOS High Sierra ...

**Comment by bcardiff** [comment]
Thanks @dustinlahr

**Comment by RX14** [comment]
Has anyone been working on fixing this packaging issue? I'd do it myself if I had a mac.

**Comment by bcardiff** [comment]
@RX14 what could be changed to delay the issue is to avoid the dependency of ssl with the playground agent. Using pipes or sockets will be enough.

Other than that, when compiling an http server eventually the user will need some ssl and pkg-config configured accordingly.

**Comment by rdp** [comment]
Something doesn't feel quite right.  Crystal is installed via a package manager.  Then it subsequently cannot use the SSL dependency that was pulled in by the same package manager.  At worst crystal should have a warning message in its brew package at install time but really this should "just work" ...

**Comment by bcardiff** [comment]
I agree, eventually crystal via brew might have a wrapper script that will set up accordings paths to link directly to OpenSSL in brew probably.

**Pull request merged**